bot/src/main/java/de/chojo/gamejam/commands/JamAdmin.java

```diff
@@ -6,36 +6,176 @@
+        var topic = String.join("\n", event.getOption("topic").getAsString(), event.getOption("topic-tagline", "", OptionMapping::getAsString));
+        ZoneId timezone;
+        try {
+            timezone = ZoneId.of(event.getOption("timezone").getAsString());
+        } catch (DateTimeException e) {
+            event.reply("Invalid timezone.").queue();
+            return;
+        }
+
+        var jamBuilder = new JamBuilder(-1)
+                .setTopic(topic);
+        try {
+            var registerStart = parseTime(event.getOption("register-start").getAsString(), timezone);
+            var registerEnd = parseTime(event.getOption("register-end").getAsString(), timezone);
+            var jamStart = parseTime(event.getOption("jam-start").getAsString(), timezone);
+            var jamEnd = parseTime(event.getOption("jam-end").getAsString(), timezone);
+            var times = new JamTimes(timezone, new TimeFrame(registerStart, registerEnd), new TimeFrame(jamStart, jamEnd));
+            jamBuilder.setTimes(times);
+        } catch (DateTimeException e) {
+            event.reply("Invalid time format. Format is YYYY.MM.DD hh:mm.").queue();
+            return;
+        }
+
+        jamData.createJam(jamBuilder.build(), event.getGuild());
+        event.reply("Jam created.").setEphemeral(true).queue();
+    }
+
+    private void startJam(SlashCommandInteractionEvent event) {
+        jamData.getActiveJam(event.getGuild()).thenAccept(jam -> {
+            if (jam.isPresent()) {
+                event.reply("A jam is already active.").setEphemeral(true).queue();
+                return;
+            }
+            var nextJam = jamData.getNextOrCurrentJam(event.getGuild()).join();
+            if (nextJam.isEmpty()) {
+                event.reply("There is no upcoming jam.").queue();
+                return;
+            }
+            nextJam.get().state().active(true);
+            jamData.updateJamState(jam.get());
+            event.reply("Jam state changed to active").queue();
+        });
+    }
 
+    private void changeVotes(SlashCommandInteractionEvent event, boolean voting, String content) {
+        jamData.getActiveJam(event.getGuild()).thenAccept(jam -> {
+            if (jam.isEmpty()) {
+                event.reply("There is no active jam.").queue();
+                return;
+            }
+            jam.get().state().voting(voting);
+            jamData.updateJamState(jam.get());
+            event.reply(content).queue();
+        });
+    }
+
+    private void endJam(SlashCommandInteractionEvent event) {
+        if (!event.getOption("confirm").getAsBoolean()) {
+            event.reply("Please confirm").queue();
+            return;
+        }
+
+        jamData.getActiveJam(event.getGuild()).thenAccept(jam -> {
+            if (jam.isEmpty()) {
+                event.reply("There is no active jam.").queue();
+                return;
+            }
+            jam.get().state().ended(true);
```

Comment:
Against law of Demeter. Should be reduced to `jam.get().finish()` which triggers all x state changes